book/src/architecure.md

```diff
@@ -0,0 +1,18 @@
+# High-Level Architecure Overview
```

Comment:
```suggestion
# High-Level Architecture Overview
```

FWIW, `mdbook serve --open` is pretty nice to develop, if you add `- [High-Level Architecture Overview](./high_level_architecture_overview.md)` to the summary, then when you save it'll detect it and auto-create this file with the same title.

And thus relatedly: I think either this title should be changed to match the one in the summary, or the one in the summary to match this one :)